### PR TITLE
Simplify version check and make gdrive loading optional

### PR DESCRIPTION
**Pull request by paplorinc**
Thanks for this awesome lib!

### CODE REVIEW

File: LLaMA_LoRA.ipynb [hunk 1]
```diff
@@ -60,20 +60,15 @@
         "# @title A small workaround { display-mode: \"form\" }\n",
         "# @markdown Don't panic if you see an error here. Just click the `RESTART RUNTIME` button in the output below, then Run All again.\n",
         "# @markdown The error will disappear on the next run.\n",
-        "!pip install Pillow==9.3.0 numpy==1.23.5\n",
+        "%pip install Pillow==9.3.0 numpy==1.23.5\n",
```
Comment (paplorinc) [inline LLaMA_LoRA.ipynb:63]: colab seems to understand this slightly better, if nothing else syntax highlighting-wise

File: LLaMA_LoRA.ipynb [hunk 2]
```diff
@@ -60,20 +60,15 @@
...
-        "print('PIL', version_float)\n",
-        "if version_float < 9.003:\n",
-        "    raise Exception(\"Restart the runtime by clicking the 'RESTART RUNTIME' button above (or Runtime > Restart Runtime).\")\n",
-        "\n",
         "import numpy\n",
-        "major, minor = map(float, numpy.__version__.split(\".\")[:2])\n",
-        "version_float = major + minor / 10**len(str(minor))\n",
-        "print('numpy', version_float)\n",
-        "if version_float < 1.0023:\n",
+        "for module, min_version in [(PIL, \"9.3\"), (numpy, \"1.23\")]:\n",
```
Comment (paplorinc) [inline LLaMA_LoRA.ipynb:68]: repeat the versions, and iterate to check if they're loaded

File: LLaMA_LoRA.ipynb [hunk 3]
```diff
@@ -60,20 +60,15 @@
...
+        "  lib_version = r.parse_version(module.__version__)\n",
+        "  print(module.__name__, lib_version)\n",
+        "  if lib_version < r.parse_version(min_version):\n",
         "    raise Exception(\"Restart the runtime by clicking the 'RESTART RUNTIME' button above (or Runtime > Restart Runtime).\")"
```
Comment (paplorinc) [inline LLaMA_LoRA.ipynb:72]: could use `exit()` as well which would "restart" it automatically, but colab doesn't seem to like it

File: LLaMA_LoRA.ipynb [hunk 4]
```diff
@@ -144,15 +139,17 @@
...
-        "!touch \"data/This folder is used by the Colab notebook \\\"{colab_notebook_filename}\\\".txt\"\n",
-        "!echo \"Data will be stored in Google Drive folder: \\\"{google_drive_data_directory_relative_path}\\\", which is mounted under \\\"{google_drive_data_directory_path}\\\"\"\n"
+        "  google_drive_data_directory_relative_path = google_drive_folder\n",
+        "  google_drive_data_directory_path = f\"{google_drive_mount_path}/My Drive/{google_drive_data_directory_relative_path}\"\n",
+        "  !mkdir -p \"{google_drive_data_directory_path}\"\n",
+        "  !ln -nsf \"{google_drive_data_directory_path}\" ./data\n",
+        "  !touch \"data/This folder is used by the Colab notebook \\\"{colab_notebook_filename}\\\".txt\"\n",
+        "  !echo \"Data will be stored in Google Drive folder: \\\"{google_drive_data_directory_relative_path}\\\", which is mounted under \\\"{google_drive_data_directory_path}\\\"\"\n",
+        "except Exception as e:\n",
+        "  print(\"Drive won't be mounted!\")"
```
Comment (paplorinc) [inline LLaMA_LoRA.ipynb:152]: let's make this part optional, it could scare people away

**Comment by zetavg** [comment]
Thank you for the PR! 💙 💜 ❤️ 🧡 💛 💚

File: LLaMA_LoRA.ipynb (hunk 4 above)
Comment (zetavg) [reply LLaMA_LoRA.ipynb:152]: I can't agree with you more.

File: LLaMA_LoRA.ipynb (hunk 3 above)
Comment (zetavg) [reply LLaMA_LoRA.ipynb:72]: This reminds me that I've been using `os.kill(os.getpid(), 9)` to restart Colab notebooks before. It actually works quite well, as only the IPython will be restarted, with all the files remaining. Just felt it was too violent to be used here...

File: LLaMA_LoRA.ipynb (hunk 2 above)
Comment (zetavg) [reply LLaMA_LoRA.ipynb:68]: Nice!

File: LLaMA_LoRA.ipynb (hunk 1 above)
Comment (zetavg) [reply LLaMA_LoRA.ipynb:63]: Hmm, I still can't tell what's the difference here. Can you give me some examples of references if you have time?

All I know about the difference between `%` and `!` is that changing the directory with `%cd` will be preserved, while `!cd` will not.

Comment (paplorinc) [reply LLaMA_LoRA.ipynb:63]: My understanding is that `%pip` is more "native", my friend said:
image

**Pull request merged**